File: length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        return len(s.strip().split(" ")[-1])

        k = 0
        i = 0
        length = 0
        while i < len(s) - 1:
            if s[i] == " " and s[i + 1] != " ":
                k = i + 1
            i += 1
        while k < len(s):
            if s[k] == " ":
                break
            length += 1
            k += 1
        return length


test_cases = ["Hello World", "   fly me   to   the moon  ", "luffy is still joyboy"]

solution = Solution()
for test_case in test_cases:
    print(f"length: {solution.lengthOfLastWord(test_case)}")

File: length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
  const word_list = s.trim().split(" ");
  return word_list[word_list.length - 1].length;

  let length = 0;
  let k = 0;
  for (let i = 0; i < s.length - 1; i++) {
    if (s[i] === " " && s[i + 1] !== " ") {
      k = i + 1;
    }
  }
  for (let i = k; i < s.length; i++) {
    if (s[i] === " ") {
      break;
    }
    length++;
  }
  return length;
};

const testCases = [
  "Hello World",
  "   fly me   to   the moon  ",
  "luffy is still joyboy",
];

for (const testCase of testCases) {
  console.log(lengthOfLastWord(testCase));
}
